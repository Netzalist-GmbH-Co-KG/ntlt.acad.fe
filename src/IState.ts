export interface ISession {
    currentUser: string,
    sessionStart: Date,
}

export interface IState {
    currentSession: ISession | null,
    applicationStart: string,
    reducerCounter: number,
}
