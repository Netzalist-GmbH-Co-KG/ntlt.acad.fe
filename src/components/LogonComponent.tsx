import * as React from 'react';

class LogonComponent extends React.Component<{onLogonRequest: (user:string, password:string) => void}, { logonMessage: string, username: string, password: string}> {

    constructor(props: Readonly<{onLogonRequest: (user:string, password:string) => void}>){
        super(props);

        this.state={ logonMessage: "", username: "", password: "" }
    }

    public render() {
        return (
            <div>
                <table>
                    <tbody>
                        <tr>
                            <td>
                                Username
                            </td>
                            <td>
                            <input id="username" type="text" onChange={ this.usernameChanged } />
                            </td>
                        </tr>
                        <tr>
                            <td>
                                Password
                            </td>
                            <td>
                                <input id="password" type="password" onChange={ this.passwordChanged } />
                            </td>
                        </tr>
                        <tr>
                            <td  colSpan={2}>
                                <button id="logonButton" onClick={ this.logonClick }>Logon</button>
                            </td>
                        </tr>
                        <tr>
                            <td  colSpan={2}>
                                <span id="logonMessage">{this.state.logonMessage}</span>
                            </td>
                        </tr>
                    </tbody>
                </table>
            </div>
        );
    }

    private usernameChanged = (ev: any) => {
        this.setState( { username: ev.target.value });
    }

    private passwordChanged = (ev: any) => {
        this.setState( { password: ev.target.value });
    }

    private logonClick = () => {
        this.setState( { logonMessage: `OMG -${this.state.username}- -${this.state.password}-` });
        this.props.onLogonRequest(this.state.username, this.state.password);
    }
}

export default LogonComponent;
