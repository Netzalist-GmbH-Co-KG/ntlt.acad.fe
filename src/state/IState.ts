export const initialState: IState = {
    demo: { 
        applicationStart: Date(),
        reducerCounter: 0
    },
    currentSession: null
}

export interface ISession {
    currentUser: string,
    sessionStart: Date,
}

export interface IDemo {
    applicationStart: string,
    reducerCounter: number,
}

export interface IState {
    currentSession: ISession | null,
    demo: IDemo
}
