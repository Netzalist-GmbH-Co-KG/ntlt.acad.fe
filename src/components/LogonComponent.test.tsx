import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { IState } from '../state/IState'
import LogonComponent from './LogonComponent';

let store:any = null;

beforeAll( () => {
  Enzyme.configure({ adapter: new Adapter() })
} )

beforeEach( () =>{
  store = createStore<IState, any, any, any>((state = { demo: { applicationStart: "2014-01-01", reducerCounter: 0}, currentSession: null }, action) => state);
});

it('renders without crashing (Smoke Test)', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Provider store={store}><LogonComponent /></Provider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders like the last time (snapshots)', () => {

  // const cb = (u:string,p:string) => { return; };
  const wrapper =  Enzyme.mount(<Provider store={store}><LogonComponent /></Provider>);

  expect(toJson(wrapper)).toMatchSnapshot();

  // click logon without entering anything 
  wrapper.find("#logonButton").simulate("click");
  expect(toJson(wrapper)).toMatchSnapshot();

  // click logon with entering someting
  wrapper.find("#username").simulate("change", { target: { value: 'user' } });
  wrapper.find("#password").simulate("change", { target: { value: 'password' } });
  wrapper.find("#logonButton").simulate("click");

  expect(toJson(wrapper)).toMatchSnapshot();

});
