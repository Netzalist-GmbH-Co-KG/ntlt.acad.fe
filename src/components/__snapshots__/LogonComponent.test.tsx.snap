// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders like the last time (snapshots) 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(LogonComponent)>
    <LogonComponent
      currentSession={null}
      tryLogon={[Function]}
    >
      <div
        className="form-signin"
      >
        <h1
          className="h3 mb-3 font-weight-normal"
        >
          Login
        </h1>
        <label
          className="sr-only"
          htmlFor="inputEmail"
        >
          Email address
        </label>
        <input
          autoFocus={true}
          className="form-control"
          id="username"
          onChange={[Function]}
          placeholder="Email address"
          required={true}
          type="text"
        />
        <label
          className="sr-only"
          htmlFor="inputPassword"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
        <button
          className="btn btn-lg btn-primary btn-block"
          id="logonButton"
          onClick={[Function]}
        >
          Sign in
        </button>
        <label
          className="badge badge-danger"
        >
          Not logged in
        </label>
      </div>
    </LogonComponent>
  </Connect(LogonComponent)>
</Provider>
`;

exports[`renders like the last time (snapshots) 2`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(LogonComponent)>
    <LogonComponent
      currentSession={null}
      tryLogon={[Function]}
    >
      <div
        className="form-signin"
      >
        <h1
          className="h3 mb-3 font-weight-normal"
        >
          Login
        </h1>
        <label
          className="sr-only"
          htmlFor="inputEmail"
        >
          Email address
        </label>
        <input
          autoFocus={true}
          className="form-control"
          id="username"
          onChange={[Function]}
          placeholder="Email address"
          required={true}
          type="text"
        />
        <label
          className="sr-only"
          htmlFor="inputPassword"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
        <button
          className="btn btn-lg btn-primary btn-block"
          id="logonButton"
          onClick={[Function]}
        >
          Sign in
        </button>
        <label
          className="badge badge-danger"
        >
          Not logged in
        </label>
      </div>
    </LogonComponent>
  </Connect(LogonComponent)>
</Provider>
`;

exports[`renders like the last time (snapshots) 3`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(LogonComponent)>
    <LogonComponent
      currentSession={null}
      tryLogon={[Function]}
    >
      <div
        className="form-signin"
      >
        <h1
          className="h3 mb-3 font-weight-normal"
        >
          Login
        </h1>
        <label
          className="sr-only"
          htmlFor="inputEmail"
        >
          Email address
        </label>
        <input
          autoFocus={true}
          className="form-control"
          id="username"
          onChange={[Function]}
          placeholder="Email address"
          required={true}
          type="text"
        />
        <label
          className="sr-only"
          htmlFor="inputPassword"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
        />
        <button
          className="btn btn-lg btn-primary btn-block"
          id="logonButton"
          onClick={[Function]}
        >
          Sign in
        </button>
        <label
          className="badge badge-danger"
        >
          Not logged in
        </label>
      </div>
    </LogonComponent>
  </Connect(LogonComponent)>
</Provider>
`;
